pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1595,7 +1607,7 @@ func (n *NetworkProvider) getNonOverlappingDefaultCIDR(ctx context.Context) (*ne
 
 			// check for overlap
 			// if it overlaps continue with the loop
-			if clusterNet.Contains(defaultNet.IP) || defaultNet.Contains(clusterNet.IP) {
+			if clusterNet.Contains(defaultNet.IP) || defaultNet.Contains(clusterNet.IP) || podCIDR.Contains(defaultNet.IP) || defaultNet.Contains(podCIDR.IP) || serviceCIDR.Contains(defaultNet.IP) || defaultNet.Contains(serviceCIDR.IP) {
```

Comment:
At a glance this is hard to read, i think we should break it out into separate blocks. 
```
	if foundClusterNet {
		if clusterNet.Contains(foo) || foo.Contains(clusterNet) {
			continue
		}
	}
	
	if foundPodNet {
		if podNet.Contains(foo) || foo.Contains(podNet) {
			continue
		}
	}
	
	if foundServiceNet {
		if serviceNet.Contains(foo) || foo.Contains(serviceNet) {
			continue
		}
	}
}
```
Bit more verbose but I think it reads better.

Reply:
what do you think about something like this 
```
     // checking clusterVpc (machine) cidr
if clusterNet.Contains(defaultNet.IP) || defaultNet.Contains(clusterNet.IP) ||
    //checking pod cidr overlap
    podCIDR.Contains(defaultNet.IP) || defaultNet.Contains(podCIDR.IP) ||
    //checking service cidr overlap
    serviceCIDR.Contains(defaultNet.IP) || defaultNet.Contains(serviceCIDR.IP) {
    continue
}
```
Easier to read but less need for the verboseness of the extra if statements

Reply:
i think we need the `found` check. In the case that the `Net` object is nil. Due to how the getters are, we could return a nill object. 
if we run the `.Contains` on a nill object we will get a panic.

Reply:
ahh ok cool